Update PR Comments & Gates with limitations (#32304)
DataDog/documentation: content/en/pr_gates/_index.md

After creating PR Gates rules, Datadog will automatically create checks on your pull requests using the [GitHub integration][13] or [Azure DevOps integration][14]. Set those checks as required in GitHub or Azure DevOps when you are ready to enforce them.

<div class="alert alert-warning">

>   PR Gates are not supported in pull requests in public repositories or that are targeting a destination branch that is in a different repository from the source branch (i.e. forked repositories trying to merge into the main repository).

Reviewer: ```suggestion
  PR Gates are not supported in pull requests in public repositories, or on pull requests targeting a destination branch in a different repository from the source branch (such as forked repositories trying to merge into the main repository).
```
